src/Lykke.Service.ExchangeDataStore/Lykke.Service.ExchangeDataStore.AzureRepositories/BaseEntity.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Table;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace Lykke.Service.ExchangeDataStore.AzureRepositories
+{
+    public class BaseEntity : TableEntity 
+    {
+        public override void ReadEntity(IDictionary<string, EntityProperty> properties, OperationContext operationContext)
+        {
+            base.ReadEntity(properties, operationContext);
+
+            foreach (var p in GetType().GetProperties().Where(x => x.PropertyType == typeof(decimal) && properties.ContainsKey(x.Name)))
```

Comment:
I fact OrderBookSnapshotEntity does not need any special processing for decimal properties because it doesn't have any - only string and DateTime props. This made BaseEntity class redundant.  I still needed  OrderBookSnapshotEntity to be Azure Entity type though, so i inherited from AzureTableEntity